crates/ruma-macros/src/events/event_type.rs

```diff
@@ -139,28 +140,30 @@ fn generate_enum(
         })
         .collect::<syn::Result<_>>()?;
 
-    let from_str_match_arms: Vec<_> = deduped
-        .iter()
-        .map(|e| {
```

Comment:
Instead of going for a nested loop, you can also change this `.map` to `.flat_map` and return an iterator of match arms from the closure.

It would still be nice not to have multiple match arms with the same body for the non-wildcard case, but that's not super important.

Reply:
How does that works with the `Err` from the `event.to_variant()` call? I would have an `Iterator` of `Result`s of `Iterator`.

The solution I have is instead of using `.flat_map`, I `.collect()` the inner iterator into a `TokenStream` and then I `.collect::<syn::Result<_>>()` the outer iterator into a `TokenStream`.

Maybe there's a more idiomatic way to do that?

Reply:
Ah, right. Perhaps something with `try_fold` instead of `flat_map` + `collect` would work. Otherwise let's keep the loop.

Reply:
I did something with `try_fold`, I don't know if it's what you were expecting.

Reply:
Looks okay. But feel free to switch back to the loop if you found it more readable.

Reply:
Well ideally we would use `try_collect()` but I don't have a strong opinion either way here.

Reply:
Hmmm maybe just go back to looping then. Probably easier to read for some ppl.